src/main/scala/Builders.scala

```diff
@@ -62,26 +88,66 @@ trait EnrichedTextBuilder {
   def withObjectTitle(x: ObjectTitle): this.type = withEnrichedText(x)
   def withPronunciation(x: Pronounce): this.type = withEnrichedText(x)
   def withVirtualLocation(x: Virtloc): this.type = withEnrichedText(x)
-  def withXml(x: NodeSeq): this.type = withContent(dataRecord(x))
-  def withText(x: String): this.type = withContent(dataRecord(x))
 
   protected def withEnrichedText[T <: EnrichedTextOption : CanWriteXML](x: T): this.type = withContent(dataRecord(x))
   protected def withContent(x: DataRecord[_]): this.type
 }
 
+trait MixedContentBuilder {
+  def withText(x: String): this.type = withContent(dataRecord(x))
+
+  /** Appends the given XML to the model object.
+    * Note that this method is _not_ type-safe!
+    * __No__ validation is performed to verify that the XML matches the expected schema..
+    *
+    * @deprecated Use multiple invocations of other methods to construct the data for a type-safe approach
+    */
+  @deprecated("Use multiple invocations of other methods for a type-safe approach", since = "3.x.3")
+  def withXml(x: NodeSeq): this.type = withContent(dataRecord(x))
+
+  protected def withContent(x: DataRecord[_]): this.type
+}
+
+/** An extension point to enable writing custom XML to NITF model objects that support extensions.
+  * This feature is supported by NITF 3.6 where the model objects contain a special field called ''any'', which acts as
+  * an extension point for provider-defined properties from other namespaces.
+  *
+  * This is an example of how it can be used (with versions 3.6.x only):
+  * {{{
+  *   import com.gu.nitf.model.builders._
+  *   import com.gu.nitf.scalaxb._
+  *   import scalaxb._
+  *
+  *   val builder = new NitfBuilder() with AnyExtensionsBuilder {
+  *     protected override def withAny(x: DataRecord[_]) = { build = build.copy(any = build.any :+ x); this }
+  *   }
+  *
+  *   val nitf = builder.withXml(<my:extension/>).build
+  *   val xml = scalaxb.toXML(nitf, None, None, toScope(Some("my") -> "http://www.example.com/my-extension"))
+  * }}}
+  */
+trait AnyContentBuilder { this: Builder[_ <: { def any: Seq[scalaxb.DataRecord[Any]] }] =>
```

Comment:
[Codacy complains](https://app.codacy.com/app/hosamaly/nitf-scala/pullRequest?prid=1474748#new-file-issues-15795245614) about the structural type expression here. However, I don't think there is anything wrong with it. We're not accessing any of the fields of the `Builder`, and the generic type is erased at runtime anyway.